### DISCUSSION TITLE
Conditional element in a RootLayout (Next 13, appdir)?

### DISCUSSION THREAD

**Discussion by mokargas** [vercel/next.js#48316] [Help]
### Summary

I'm confused about RootLayouts and what approach to take. I have a simple RootLayout that contains a basic Footer element. The footer element contains some links and the like that I'd like to retain server-side.

Now consider the case that for a presentational page (in my case, my initial or home page), I would like to not render this Footer component or add a style or class to make it visually disappear. I can't seem to do this, as this layout isn't re-rendered on navigation changes, so for instance I can't add a class as it will apply to all page layouts.

I can render a client side version of the Footer, but it's my understanding this won't work with SSG. I can also create sub-layouts for every page and manually include Footer on each of them, but to me this seems redundant, as I only want it excluded from one page and then have to go into making identical nested sublayouts for little reason.

What's the approach for solving this?

**Comment by wwwdepot** [comment]
This is where Route Groups come to rescue
https://beta.nextjs.org/docs/routing/defining-routes#route-groups

```
/app
    /layout.js <-- root layout
    (landing)
        /layout.js    <-- layout w/o footer
        /page.js      <-- this is `/` home page
    (others)
       /layout.js    <-- layout with footer
       /some-path
              /page.js <-- this is `/some-path`
```

**Reply by mokargas** [reply]
@wwwdepot Thanks, this semi-works. 

It still requires the duplication of most of the layout markup for route-group layouts. It also exhibits a very very visible FOUC everytime that layout boundary is changed (e.g, changing the route from landing page, to one of the others), at least on DEV.
